recipe/conda_build_config.yaml

```diff
@@ -4,42 +4,47 @@ c_stdlib_version:  # [osx and x86_64]
 MACOSX_SDK_VERSION:         # [osx]
   - 12.3                    # [osx]
 
-# Needs higher stdlib for CUDNN9
-# CUDA 12 on aarch needs newer glibc, which requires us to override
-# the whole monster zip just to bump c_stdlib_version / docker_image
-c_stdlib_version:                                   # [linux and aarch64]
-  - 2.17                                            # [linux and aarch64]
-  - 2.17                                            # [linux and aarch64]
-  - 2.28                                            # [linux and aarch64]
-docker_image:                                       # [linux and aarch64]
-  - quay.io/condaforge/linux-anvil-cos7-x86_64      # [linux and aarch64]
-  - quay.io/condaforge/linux-anvil-cuda:11.8        # [linux and aarch64]
-  - quay.io/condaforge/linux-anvil-alma-x86_64:8    # [linux and aarch64]
-# rest just to make smithy happy
-c_compiler_version:                                 # [linux and aarch64]
-  - 13                                              # [linux and aarch64]
-  - 11                                              # [linux and aarch64]
-  - 12                                              # [linux and aarch64]
-cxx_compiler_version:                               # [linux and aarch64]
-  - 13                                              # [linux and aarch64]
-  - 11                                              # [linux and aarch64]
-  - 12                                              # [linux and aarch64]
-fortran_compiler_version:                           # [linux and aarch64]
-  - 13                                              # [linux and aarch64]
-  - 11                                              # [linux and aarch64]
-  - 12                                              # [linux and aarch64]
-cdt_name:                                           # [linux and aarch64]
-  - cos7                                            # [linux and aarch64]
-  - cos7                                            # [linux and aarch64]
-  - cos7                                            # [linux and aarch64]
-cuda_compiler:                                      # [linux and aarch64]
-  - None                                            # [linux and aarch64]
-  - nvcc                                            # [linux and aarch64]
-  - cuda-nvcc                                       # [linux and aarch64]
-cuda_compiler_version:                              # [linux and aarch64]
-  - None                                            # [linux and aarch64]
-  - 11.8                                            # [linux and aarch64]
-  - 12.0                                            # [linux and aarch64]
+c_compiler_version:         # [osx]
+  - 17                      # [osx]
+cxx_compiler_version:       # [osx]
+  - 17                      # [osx]
+llvm_openmp:                # [osx]
+  - 17                      # [osx]
+
+mkl:
+  - 2023
+libprotobuf:
+  - 5.28.2
```

Comment:
This PR is mainly to keep building 2.4 as before. However, we could remove these bits in another PR and see what a rerender brings (there were some issues with clang 18 in the fbgemm submodule, but there have been no changes in that submodule for ~2 years, so no idea what would have changed between 2.4 & 2.5, where we're successfully using clang 18.